### PR TITLE
enable SnapStart with Java 11 runtime

### PR DESCRIPTION
**Pull request by bnusunny**
Use Java 11 runtime and enable SnapStart

### CODE REVIEW

File: infrastructure/cdk/src/main/java/com/unicorn/UnicornStoreSpringStack.java [hunk 1]
```diff
@@ -35,32 +40,31 @@ public UnicornStoreSpringStack(final Construct scope, final String id, final Sta
                 .build());
     }
 
-    private RestApi setupRestApi(Function unicornStoreSpringLambda) {
+    private RestApi setupRestApi(IFunction hanlder) {
         return LambdaRestApi.Builder.create(this, "UnicornStoreSpringApi")
                 .restApiName("UnicornStoreSpringApi")
-                .handler(unicornStoreSpringLambda)
+                .handler(hanlder)
```
Comment (msailes) [inline infrastructure/cdk/src/main/java/com/unicorn/UnicornStoreSpringStack.java]: typo: handler

File: infrastructure/cdk/src/main/java/com/unicorn/UnicornStoreSpringStack.java [hunk 2]
```diff
@@ -35,32 +40,31 @@ public UnicornStoreSpringStack(final Construct scope, final String id, final Sta
                 .build());
     }
 
-    private RestApi setupRestApi(Function unicornStoreSpringLambda) {
+    private RestApi setupRestApi(IFunction hanlder) {
```
Comment (msailes) [inline infrastructure/cdk/src/main/java/com/unicorn/UnicornStoreSpringStack.java]: typo: handler

File: infrastructure/cdk/src/main/java/com/unicorn/UnicornStoreSpringStack.java (hunk 1 above)
Comment (bnusunny) [reply infrastructure/cdk/src/main/java/com/unicorn/UnicornStoreSpringStack.java]: Corrected.

File: infrastructure/cdk/src/main/java/com/unicorn/UnicornStoreSpringStack.java (hunk 2 above)
Comment (bnusunny) [reply infrastructure/cdk/src/main/java/com/unicorn/UnicornStoreSpringStack.java]: Corrected.

**Comment by maschnetwork** [comment]
LGTM

**Pull request merged**